src/main/ruby/truffleruby/core/encoding.rb

```diff
@@ -139,6 +139,8 @@ def names
   end
 
   def replicate(name)
+    warn 'Encoding#replicate is deprecated and will be removed in Ruby 3.3; use the original encoding instead', category: :deprecated, uplevel: 2
```

Comment:
```suggestion
    warn 'Encoding#replicate is deprecated and will be removed in Ruby 3.3; use the original encoding instead', category: :deprecated, uplevel: 1
```

We need a line with `Encoding#replicate` method call to be mentioned in the warning, so it's 1 level above the current Ruby frame